templates/skeleton/app/entry.server.tsx

```diff
@@ -2,16 +2,22 @@ import type {EntryContext} from '@shopify/remix-oxygen';
 import {RemixServer} from '@remix-run/react';
 import isbot from 'isbot';
 import {renderToReadableStream} from 'react-dom/server';
+import {NonceProvider} from './utils';
 
 export default async function handleRequest(
   request: Request,
   responseStatusCode: number,
   responseHeaders: Headers,
   remixContext: EntryContext,
 ) {
+  const nonce = (request as any).nonce as string;
+
   const body = await renderToReadableStream(
-    <RemixServer context={remixContext} url={request.url} />,
+    <NonceProvider value={nonce}>
```

Comment:
The `<NonceProvider>` is defined only for `entry.server.tsx`, not in `entry.client.tsx`, which means that any client code using the `useNonce()` hook will get the default undefined value. This is important so that the nonce value isn't scrapeable via a browser extension.